tests/internal/utils.c

```diff
@@ -147,6 +175,101 @@ void test_write_str()
     TEST_CHECK(ret == FLB_FALSE);
 }
 
+void test_write_str_invalid_trailing_bytes()
+{
+    struct write_str_case cases[] = {
+        /* Invalid unicode (one bad trailing byte) */
+        {
+            "\xe3\x81\x01""abc", 6,  /* note that 0x01 is an invalid byte */
+            "�""\\u0001abc"  /* replace invalid unicode */
```

Comment:
nitpick: I'm not sure if there's anything in the coding style guide about unicode in the source code but I personally would prefer is there was no unicode in the source code, not because of the latest trend but because I think it pollutes the files.

Reply:
Sure, this can be changed into escaped version of the replacement character.